### ISSUE TITLE
Maine Baseflow Layer Updates

### ISSUE THREAD

**Issue by harper-wavra** [USGS-WiM/StreamStats#1301]
Fixes for #1275

- [x] 1) The legend values have been updated to always have the same min/max values, but the legend shown in StreamStats is not reflecting this change. Is there a way to use the [ArcGIS API legend widget](https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdevelopers.arcgis.com%2Fjavascript%2Flatest%2Fapi-reference%2Fesri-widgets-Legend.html%23%3A~%3Atext%3DSince%253A%2520ArcGIS%2520API%2520for%2520JavaScript%25204.0%2520The%2520Legend%2Care%2520configured%2520in%2520the%2520Renderer%2520of%2520the%2520layer.&data=05%7C01%7Chwavra%40usgs.gov%7C45853fef655c40e8546908da29dd7612%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637868328818870021%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C2000%7C%7C%7C&sdata=MFFa6DssxLkiVPrP1%2FGgOnXNnRgX24BEZ5osMvkgapc%3D&reserved=0) so that it will always reference the published layer service legend? If not, simply changing the values for each view level in your code would be fine. I do not expect the legend to change again, but I'll keep this in mind if it does.

- [x] 2) Ideally it would be nice to show the layer starting at zoom level 9 or 10, but I'm guessing the issue you are having with rendering will not allow this. Maybe we can find another option where we host a vector tile service which is much faster for rendering at each view, but use the feature service to view pop-up information, but not actually render the feature service in the map if that makes sense. Unfortunately, they do not have support to show pop-up information for the vector tile service yet because it is essentially just georeferenced jpegs. Here is a link to the [vector tile service](https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fusgs.maps.arcgis.com%2Fhome%2Fitem.html%3Fid%3De814195aa5364e2c9c862102a641a6f8&data=05%7C01%7Chwavra%40usgs.gov%7Ca441238c4203488d13d808da29fd939e%7C0693b5ba4b184d7b9341f32f400a5494%7C0%7C0%7C637868466925174725%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C2000%7C%7C%7C&sdata=xc40vtMvq1HV2xx2VPmMY%2BNvqhh3TZdHYsdE9cClyiI%3D&reserved=0) for you to try. The tricky part would be to show the correct legend when the view changes, because another limitation to this type of service is that it does not have a legend. I'm assuming that you are still not able to get the state and county views to show at zoom levels <12? It's helpful to see an overview of the layer at these levels to see the variation in baseflow symbology, but if it's just not rendering at these levels then maybe we should open another ticket with ESRI to see what the issue is because it does not seem to be a problem in AGOL web apps or ArcGIS Pro.

- [x] 3) Is it possible to widen the query click point tolerance at each zoom level to be similar to how the gage point query responds? Querying a gage point is much more forgiving, but the tolerance for the baseflow line features is too precise only allowing for pop-up information on the baseflow layer at zoom levels >=14.

**Comment by harper-wavra** [comment]
Not sure if we can use the ArcGIS API legend widget as that seems to create is own legend and we are using the legend from wim_angular. I just updated the hard coded values for now as Luke said they shouldn't change anymore, but maybe we can come back to getting them on the fly when time allows / if we find the values are actually are changing.

**Comment by amedenblik** [comment]
Item number 2 does not seem possible with the current Feature Service or Vector Tile Service. The map service will need to be reconfigured to be compatible with StreamStats infrastructure. Awaiting response from Luke but will likely close this issue without completing item number 2 until a new map service is provided.

**Comment by harper-wavra** [comment]
Try to use new map published on sciencebase: https://www.sciencebase.gov/catalog/item/628bc405d34e4fef2ec3d50c

**Comment by harper-wavra** [comment]
In the appconfig if you replace the ME layer with:
```
{
            "MeanAugustBaseflow": {
                "name": "Mean August Baseflow",
                "url": "https://gis.usgs.gov/sciencebase3/rest/services/Catalog/628bc405d34e4fef2ec3d50c/MapServer",
                "type": 'agsDynamic',
                "visible": true,
                "layerOptions": {
                    "zIndex": 1,
                    "format": "png",
                    "layers": [0],
                    "f": "image",
                    "minZoom": 9
                },
                "queryProperties": { 
                    "Mean August Baseflow": { 
                        "GNIS_Name": "GNIS Name", 
                        "DASQMI": "Drainage Area (mi2)",
                        "SANDGRAVAF": "Aquifer Area (%)",
                        "JULYAVPRE": "Mean July Precip (in)",
                        "AUGAVGBF": "Mean August Baseflow (cfs/mi2",
                        "OOB_DA": "Drainage Area out-of-bounds",
                        "OOB_JULYAV":"Mean July Precip out-of-bounds",
                        "OOB_WARNIN": "% Aquifer Area out-of-bounds",
                        "REGULATED": "Regulated stream/river"
                    } 
                }
            }
        }
```

it will load at a zoom level of 9. We will have to rework how to get the popups and legend to show correctly though. 

Waiting on a response from Luke.

**Comment by harper-wavra** [comment]
Luke said he is good with the sciencebase WMS performance. We will need to:
- [x] Edit the appconfig
- [x] Get the legend working
- [x] Get the popups working - 0 is false 1 is true
- [x] Remove all unused code from this issue and issue #1275. See https://github.com/USGS-WiM/StreamStats/pull/1298


In the branch HW-1301, I have the map working at a zoom level of 9 or greater and I think the popups and legend text is working. Just need to test it out some more.
